pkg/core/statesync/mptpool.go

```diff
@@ -48,6 +49,26 @@ func (mp *Pool) GetAll() map[util.Uint256][][]byte {
 	return mp.hashes
 }
 
+// GetBatch returns set of unknown MPT nodes hashes (MaxMPTHashesCount at max).
+func (mp *Pool) GetBatch() map[util.Uint256]struct{} {
+	mp.lock.RLock()
+	defer mp.lock.RUnlock()
+
+	count := len(mp.hashes)
+	if count > payload.MaxMPTHashesCount {
+		count = payload.MaxMPTHashesCount
```

Comment:
This creates a dependency on `payload` package which is trivial to avoid if you're to make it `GetBatch(limit int)`.

Reply:
Fixed in a separate commit.